src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -70,6 +72,48 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
         )
         return extension, name, create_identity
 
+    def Delete(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, yes):
+        # Delete DCR-A if it exists incase of MSI Auth
+        useAADAuth = False
+        isDCRAExists = False
+        cluster_rp, _ = get_cluster_rp_api_version(cluster_type)
+        try:
+            extension = client.get(resource_group_name, cluster_rp, cluster_type, cluster_name, name)
+        except Exception:
+            pass  # its OK to ignore the exception since MSI auth in preview
+
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        # handle cluster type here
+        cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/{2}/{3}/{4}'.format(subscription_id, resource_group_name, cluster_rp, cluster_type, cluster_name)
+        if (extension is not None) and (extension.configuration_settings is not None):
+            configSettings = extension.configuration_settings
+            if 'omsagent.useAADAuth' in configSettings:
+                useAADAuthSetting = configSettings['omsagent.useAADAuth']
+                if (isinstance(useAADAuthSetting, str) and str(useAADAuthSetting).lower() == "true") or (isinstance(useAADAuthSetting, bool) and useAADAuthSetting):
+                    useAADAuth = True
+        if useAADAuth:
+            association_url = cmd.cli_ctx.cloud.endpoints.resource_manager + f"{cluster_resource_id}/providers/Microsoft.Insights/dataCollectionRuleAssociations/ContainerInsightsExtension?api-version=2019-11-01-preview"
+            for _ in range(3):
+                try:
+                    send_raw_request(cmd.cli_ctx, "GET", association_url,)

```

Comment:
Why don't we call SDK methods instead of requesting these REST directly? Is it because the SDK does not support related methods at present?

Reply:
@ganga1980 Can you comment on this one?

Reply:
> Why don't we call SDK methods instead of requesting these REST directly? Is it because the SDK does not support related methods at present?

> We have used same implementation for AKS & Arc-K8s Monitoring. When originally, implemented for AKS, SDK didnt have support for these methods and hence implemented via REST APIs directly.  I will explore and implement these methods using SDK for our GA release.

Reply:
Resolving this one @zhoxing-ms since @ganga1980 will fix in later release.